stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -117,7 +117,11 @@ internal class PaymentOptionsActivity : BasePaymentSheetActivity<PaymentOptionRe
         viewModel.fetchFragmentConfig().observe(this) { config ->
             if (config != null) {
                 viewModel.transitionTo(
-                    if (starterArgs.paymentMethods.isEmpty() && starterArgs.newCard == null) {
+                    // It would be nice to see this condition move into the PaymentOptionsListFragment
+                    // where we also jump to a new unsaved card.  However this move require
+                    // the transition target to specify when to and when not to add things to the
+                    // backstack.
+                    if (starterArgs.paymentMethods.isEmpty()) {
```

Comment:
If the paymentMethods is empty we need to jump to the add card because the new card won't be showing in the list.